### DISCUSSION TITLE
Web server Deployment (Self Hosting)

### DISCUSSION THREAD

**Discussion by Godmemer420** [flet-dev/flet#1507] [Q&A] [answered]
### Question

So I've got an Ubuntu machine and I tried running a flet web server on it. Open the main.py in webview, put in port, reverse proxy it through nginx. It works fine and I can access the webserver from anywhere BUT no matter on which device I open the web server or just 2 different tabs then variables that I have defined stay stored. I mean I don't see what someone else is typing when opening google but with my implementation I currently do. I heard that I can use gunicorn but that just didn't worked as it couldn't find app (normally you use gunicorn for flask servers). If anyone knows what I can use then feel free to suggest.

EDIT: Anwser is that I should have declared that global variables as page.VARIABLE instead of just globally.

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by FeodorFitsner** [comment, marked as answer]
It could be a problem with your program, i.e. created control are not unique for the given user session.

**Reply by Godmemer420** [reply]
I actually found out the problem thanks to you comment as I looked at it from a different perspective thanks to you! Thank you for your help!
